CAIPs/caip-307.md

```diff
@@ -0,0 +1,88 @@
+---
+caip: 307
+title: JSON-RPC Method for Retrieving Session Authorizations
+author: [Alex Donesky] (@adonesky1)
+discussions-to: TBD
+status: Draft
+type: Standard
+created: 2024-07-13
+requires: 25, 217
+---
+
+## Simple Summary
+
+CAIP-307 introduces the `wallet_getSession` method for retrieving authorizations from an active [CAIP-25][] initiated session.
+
+## Abstract
+
+This proposal aims to extend the [CAIP-25][] standard by defining a new JSON-RPC method for retrieving authorizations within a session. This method allows callers to dynamically retrieve authorizations and properties without necessarily having to persist and track it throughout the session's life.
+
+## Motivation
+
+The motivation behind this proposal is to enhance the flexibility of [CAIP-25][] by enabling the retrieval of session authorizations at any time. The proposed method provides an intuitive way to retrieve authorizations for an active session, allowing callers to access session data without having to persist and track it over the full life of the method.
+
+## Specification
+
+### Definition
+
+The `wallet_getSession` method returns an active session. If a `sessionId` is provided, it returns the authorizations for that specific session; otherwise, it returns the authorizations for the single active session between the wallet and the caller.`
```

Comment:
what if no sessionId is provided in the request, but the wallet has multiple sessions concurrently?  I.e., what if the initial connections all returned a sessionId but the dapp just messed up and didn't include? It might help to be more explicit, i.e. 

```suggestion
The `wallet_getSession` method returns an active session. If a `sessionId` is provided, it returns the authorizations for that specific session; if none is provided and only one session is active, it returns the authorizations for that active session between the wallet and the caller, otherwise, an appropriate error message.
```

Reply:
done with some modifications to your suggestion here: https://github.com/ChainAgnostic/CAIPs/pull/285/commits/405c5a03d7d0386c4f3d87bcf8db8de81af6257f

```
The `wallet_getSession` method returns an active session.
If a `sessionId` is provided, it returns the authorizations for that specific session;
If no `sessionId` parameter is provided - and there is a single active session with no `sessionId` assigned - it returns the session authorizations and properties for that session;
otherwise, an appropriate error message;
```